tests/unit/pages/Editor.spec.js

```diff
@@ -103,4 +121,867 @@ describe("Editor.vue", () => {
+      showPublishedPlioDialog: true,
+      videoId: "jdYJf_ybyVo",
+    });
+
+    await wrapper
+      .find('[data-test="closePublishedPlioDialogButton"]')
+      .trigger("click");
+    expect(closePublishedPlioDialog).toHaveBeenCalled();
+    expect(wrapper.vm.showPublishedPlioDialog).toBeFalsy();
+  });
+
+  it("redirects to dashboard when analyse button is clicked", async () => {
+    const redirectToDashboard = jest.spyOn(
+      Editor.methods,
+      "redirectToDashboard"
+    );
+    const mockRouter = { push: jest.fn() };
+    const wrapper = mount(Editor, {
+      global: {
+        mocks: {
+          $router: mockRouter,
+        },
+      },
+    });
+    await wrapper.setData({
+      status: "published",
+    });
+    await wrapper.find('[data-test="analyseButton"]').trigger("click");
+    expect(redirectToDashboard).toHaveBeenCalled();
+    expect(mockRouter.push).toHaveBeenCalledWith({
+      name: "Dashboard",
+      params: {
+        org: "",
+        plioId: "",
+      },
+    });
+  });
+
+  it("deletes linked image properly", async () => {
+    const deleteLinkedImage = jest.spyOn(Editor.methods, "deleteLinkedImage");
+    const wrapper = mount(Editor);
+
+    const dummyItemsWithImage = dummyItemsCopy.data;
+    dummyItemsWithImage[0].details.image = {
+      id: 56,
+      url: "https://plio-prod-assets.s3.amazonaws.com/images/hxojrjdasf.png",
+      alt_text: "Image",
+      created_at: "2021-07-02T12:58:41.683683Z",
+      updated_at: "2021-07-02T12:58:41.684174Z",
+    };
+
+    await wrapper.setData({
+      showImageUploaderDialog: true,
+      items: dummyItemsWithImage,
+      itemImage: dummyItemsWithImage[0].details.image.url,
+      currentItemIndex: 0,
+    });
+
+    await wrapper
+      .find('[data-test="imageUploaderDialog"]')
+      .find('[data-test="deleteImageButton"]')
+      .trigger("click");
+    expect(deleteLinkedImage).toHaveBeenCalled();
+  });
+
+  it("uploads image properly via ImageUploaderDialog", async () => {
+    mockAxios.reset();
+    const uploadImage = jest.spyOn(Editor.methods, "uploadImage");
+    const submitImage = jest.spyOn(ImageUploaderDialog.methods, "submitImage");
+    const wrapper = mount(Editor);
+    await wrapper.setData({
+      items: dummyItemsCopy.data,
+      showImageUploaderDialog: true,
+      currentItemIndex: 0,
+    });
+
+    let imageUploaderWrapper = wrapper.findComponent(ImageUploaderDialog);
+    imageUploaderWrapper.setData({
+      localImageData: imageData,
+    });
+
+    await imageUploaderWrapper
+      .find('[data-test="submitImageButton"]')
+      .trigger("click");
+
+    expect(submitImage).toHaveBeenCalled();
+    expect(uploadImage).toHaveBeenCalled();
+
+    expect(mockAxios.post).toHaveBeenCalledTimes(1);
+    expect(mockAxios.post).toHaveBeenCalledWith(
+      `/images/`,
+      mockAxios.lastReqGet().data
+    );
+
+    mockAxios.mockResponse(
+      {
+        data: "mock response",
+      },
+      mockAxios.lastReqGet()
+    );
+
```

Comment:
shouldn't we also check if `/items` is being updated with the image?

Reply:
That is being checked